File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/inputs/inputWithDefaultCurrency/expected/types/OrderFilter.java
package com.netflix.graphql.dgs.codegen.java.testcases.inputs.inputWithDefaultCurrency.expected.types;

import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Currency;
import java.util.Objects;

public class OrderFilter {
  private Currency value = Currency.getInstance("USD");

  public OrderFilter() {
  }

  public OrderFilter(Currency value) {
    this.value = value;
  }

  public Currency getValue() {
    return value;
  }

  public void setValue(Currency value) {
    this.value = value;
  }

  @Override
  public String toString() {
    return "OrderFilter{value='" + value + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    OrderFilter that = (OrderFilter) o;
    return Objects.equals(value, that.value);
  }

  @Override
  public int hashCode() {
    return Objects.hash(value);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private Currency value = Currency.getInstance("USD");

    public OrderFilter build() {
      OrderFilter result = new OrderFilter();
      result.value = this.value;
      return result;
    }

    public Builder value(Currency value) {
      this.value = value;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/inputWithDefaultCurrency/expected/types/OrderFilter.kt
package com.netflix.graphql.dgs.codegen.cases.inputWithDefaultCurrency.expected.types

import com.fasterxml.jackson.`annotation`.JsonCreator
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.netflix.graphql.dgs.codegen.GraphQLInput
import java.util.Currency
import kotlin.Any
import kotlin.Pair
import kotlin.String
import kotlin.collections.List

public data class OrderFilter @JsonCreator constructor(
  @JsonProperty("value")
  public val `value`: Currency = default<OrderFilter, Currency>("value",
      java.util.Currency.getInstance("USD")),
) : GraphQLInput() {
  override fun fields(): List<Pair<String, Any?>> = listOf("value" to `value`)
}
